envi/archs/amd64/regs.py

```diff
@@ -34,18 +32,27 @@
     # Segment registers
     ("es", 16),("cs",16),("ss",16),("ds",16),("fs",16),("gs",16),
     # FPU Registers
-    ("st0", 128),("st1", 128),("st2", 128),("st3", 128),("st4", 128),("st5", 128),("st6", 128),("st7", 128),
```

Comment:
So this was a bit of an odd point. In the intel manual they're described as 80 bit registers in the x87 execution environment section of the manual. The user fpregs struct (in /usr/include/sys/user.h) for i386 reflects this in the st_space being an 20 length array of ints for 8, 80 bit registers. amd64 allocates a 32 length array of ints for it, but it looks like they shove a few other pieces of info there (need to confirm this).

Reply:
i was wondering if it may have been a typo somewhere, since 128 == 0x80.  
consequently, 128bits is 16 bytes...  and 8 of them would take up 128 bytes.  32*4 = 128.

i would not change the size of these without being absolutely sure.

Reply:
So everything I can find, even the intel manual, refers to the 80 bit wide fp stack registers. 

Also it took me way to long to make this connection, but I think  it's just dumping the xsave data for the 64 bit version. Because technically the ST space in the xsave data is 128 bits per st register, for a total of 128 bytes.

However they also specify the upper 48 bits/6 bytes of each st register space is reserved when you retrieve the xsave data, and you can't properly parse a float from it if you keep those bits in. They're still present in the data that ptrace returns, you just have to completely discount them (see section 13.4.1 of the SDM). I'm thinking that's what this is.

Also gdb only ever returns up to 10 bytes for a float when examining the st registers. So at least empirically it tracks